commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -1,349 +1,242 @@
+                cartDeliveryGroup.getDeliverToAddress().getCountry(), cart.getTaxType());
+
+                taxDataShippingItems = getTaxesFromStaticResponse(
+                chargeItemByDeliveryGroupIdMap, cartDeliveryGroup.getDeliverToAddress().getState(),
+                cartDeliveryGroup.getDeliverToAddress().getCountry(), cart.getTaxType());
             }
-        }
-      }
-
-      // If there are shipping items, add tax for them as well
-      for (String cartItemId : dataFromExternalServiceForShippingItems.keySet()) {
-        TaxDataFromExternalService taxDetailsToCartId = dataFromExternalServiceForShippingItems.get(cartItemId);
-        CartExtension.CartItem cartItem = shippingItemById.get(cartItemId);
-        boolean isCartItemModified = false;
-        // If there is any modification in unit price, delete existing and create new cart tax entries in cart item.
-        if (cartItem.getNetUnitPrice() != null &&
-          cartItem.getNetUnitPrice() != taxDetailsToCartId.getNetUnitPrice()) {
-            cartItem.getCartTaxes().remove(cartItem.getCartTaxes().get(0));
-            isCartItemModified = true;
-        }
 
-        if (cartItem.getCartTaxes() == null ||
-          cartItem.getCartTaxes().isEmpty() ||
-          isCartItemModified == true) {
-            cartItem.setNetUnitPrice(taxDetailsToCartId.getNetUnitPrice());
-            cartItem.setGrossUnitPrice(taxDetailsToCartId.getGrossUnitPrice());
-            CartExtension.CartTaxList cartTaxCollection = cartItem.getCartTaxes();
-            CartExtension.CartTax cartTax = new CartExtension.CartTax(
-              CartExtension.TaxTypeEnum.ESTIMATED,
-              taxDetailsToCartId.getAmount(),
-              taxDetailsToCartId.getTaxName()
-            );
-            cartTax.setTaxRate(String.valueOf(taxDetailsToCartId.getRate()));
-            cartTaxCollection.add(cartTax);
+            // TODO need to add cvo in case of tax data is null
+            if (taxData == null) {
+                CartExtension.CartValidationOutput cvo = new CartExtension.CartValidationOutput(
+                CartExtension.CartValidationOutputTypeEnum.TAXES,
+                CartExtension.CartValidationOutputLevelEnum.ERROR);
+                cvo.setMessage('No tax rates configured.');
+                cartValidationOutputCollection.add(cvo);
+                return;
+            }
+
+            // If no tax details are returned for any cart item, add a cart validation output entry. If
+            // any invalid scenario found then return.
+            boolean isCvoPresent = false;
+            for (String cartItemId : cartItemByIdMap.keySet()) {
+            	TaxData taxDetails = taxData.get(cartItemId);
+            	if (taxDetails == null) {
+            		CartExtension.CartValidationOutput cvo = new CartExtension.CartValidationOutput(
+            		CartExtension.CartValidationOutputTypeEnum.TAXES,
+            		CartExtension.CartValidationOutputLevelEnum.INFO);
```

Comment:
Just checking if this is informed choice to use INFO level here?

Reply:
changed to error